src/components/registration/SignUp.js

```diff
@@ -53,37 +45,28 @@ class SignUp extends Component {
     axios
       .post(`${REACT_APP_SVR_API}/user/`, data)
       .then(response => {
-        // console.log(response);
-        // Stores token in local storeage for the time being
         localStorage.setItem('app-token', response.data.token);
-        // Sweet Alert for successful registration
         handleSignup();
       })
       .catch(error => {
-        try {
-          // Handles errors that are not HTTP specific
-          console.error(error);
-          this.setState({ showRegistrationFailure: true });
-          if (!error.status) {
-            console.error('A network error has occured.');
-          } else if (error.response.status === 400) {
-            console.error('Bad Request');
-          } else if (error.response.status === 500) {
-            console.error('Something bad happended on the server.');
-          } else {
-            console.error('An unknown error has occurred');
-          }
-        } catch (ex) {
-          alert('Something went wrong...');
-          Promise.reject(ex);
+        if (error.message === 'Network Error') {
+          this.setState({
+            HTTPErrorMessage:
+              'A network error has occurred while contacting our servers...',
+          });
+        } else if (error.response.status === 400) {
+          this.setState({ HTTPErrorMessage: `Error: ${error.response.data}` });
+        } else if (error.response.status === 500) {
+          this.setState({
+            HTTPErrorMessage: `Something went wrong on our side. Please try again at a later time.\n Error: ${error.response.data}`,
+          });
         }
       });
   };
 
   // Handles getting values
   handleEmailChange = evt => {
-    const email = evt.target.value;
-    this.setState({ email });
+    this.setState({ email: evt.target.value });
```

Comment:
Same as Log In, all of these `handle______Change` functions should also reset the corresponding error and HTTP message

```suggestion
    this.setState(prevState => ({ 
      email: evt.target.value, 
      errors: { ...prevState.errors, email: false }, 
      HTTPErrorMessage: '', 
    }));
```